include/micm/util/matrix.hpp

```diff
@@ -163,7 +165,7 @@ namespace micm
                   // check that this row matches the expected rectangular matrix dimensions
                   if (other[x].size() != y_dim)
                   {
-                    throw std::runtime_error("Invalid vector for matrix assignment");
+                    throw std::system_error(make_error_code(MicmMatrixErrc::InvalidVector), "");
```

Comment:
Might be worth having a message why it's invalid?

Reply:
done!